packages/interface/src/components/ImageUpload.tsx

```diff
@@ -1,82 +1,94 @@
-import { type ComponentProps, useRef, useCallback } from "react";
+import { type ComponentProps, ElementType, forwardRef, useRef, useCallback } from "react";
 import { Controller, useFormContext } from "react-hook-form";
 import { toast } from "sonner";
 
 import { IconButton } from "~/components/ui/Button";
 
+import { PolymorphicRef } from "./ui";
+
 export interface IImageUploadProps extends ComponentProps<"img"> {
   name?: string;
   maxSize?: number;
 }
 
-export const ImageUpload = ({
-  name = "",
-  maxSize = 1024 * 1024, // 1 MB
-  className,
-}: IImageUploadProps): JSX.Element => {
-  const ref = useRef<HTMLInputElement>(null);
-  const { control } = useFormContext();
-
-  const select = useMutation({
-    mutationFn: async (file: File) => {
-      if (file.size >= maxSize) {
-        toast.error("Image too large", {
-          description: `The image to selected is: ${(file.size / 1024).toFixed(2)} / ${(maxSize / 1024).toFixed(2)} kb`,
-        });
-        throw new Error("IMAGE_TOO_LARGE");
-      }
+export const ImageUpload = forwardRef(
+  (
+    {
+      name = "",
+      maxSize = 1024 * 1024, // 1 MB
+      className,
+    }: IImageUploadProps,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    ref: PolymorphicRef<ElementType>,
+  ): JSX.Element => {
+    const internalRef = useRef<HTMLInputElement>(null);
+    const { control } = useFormContext();
 
-      return Promise.resolve(URL.createObjectURL(file));
-    },
-  });
+    const select = useMutation({
+      mutationFn: async (file: File) => {
+        if (file.size >= maxSize) {
+          toast.error("Image too large", {
+            description: `The image to selected is: ${(file.size / 1024).toFixed(2)} / ${(maxSize / 1024).toFixed(2)} kb`,
+          });
+          throw new Error("IMAGE_TOO_LARGE");
+        }
 
-  const onClickIconButton = useCallback((e: React.MouseEvent<HTMLButtonElement>) => {
-    e.preventDefault();
-  }, []);
+        return Promise.resolve(URL.createObjectURL(file));
+      },
+    });
 
-  return (
-    <Controller
-      control={control}
-      name={name}
-      render={({ field: { value, onChange, ...field } }) => (
-        // eslint-disable-next-line jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events
-        <div
-          className={clsx("relative cursor-pointer overflow-hidden", className)}
-          onClick={() => ref.current?.click()}
-        >
-          <IconButton className="absolute bottom-1 right-1" icon={ImageIcon} onClick={onClickIconButton} />
+    const onClickIconButton = useCallback((e: React.MouseEvent<HTMLButtonElement>) => {
+      e.preventDefault();
+    }, []);
 
+    return (
+      <Controller
+        control={control}
+        name={name}
+        render={({ field: { value, onChange, ...field } }) => (
+          // eslint-disable-next-line jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events
```

Comment:
div shouldn't have `onClick` handlers.

Reply:
This div contains the upload image button and the div icon. I tried to change `<div>` to `<button>` but it didn't work (we cannot have nested buttons)